src/gui/qgsmaptoolcapture.cpp

```diff
@@ -1401,19 +1403,19 @@ void QgsMapToolCapture::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
 
           if ( hasCurvedSegments && providerSupportsCurvedSegments )
           {
-            curveToAdd = captureCurve()->clone();
+            curveToAdd.reset( captureCurve()->clone() );
           }
           else
           {
-            curveToAdd = captureCurve()->curveToLine();
+            curveToAdd.reset( captureCurve()->curveToLine() );
           }
         }
         else
         {
-          curveToAdd = captureCurve()->clone();
+          curveToAdd.reset( captureCurve()->clone() );
         }
         QgsCurvePolygon *poly = new QgsCurvePolygon();
-        poly->setExteriorRing( curveToAdd );
+        poly->setExteriorRing( curveToAdd.release() );
```

Comment:
shouldn't poly in the line below be cloned, to be consistent with ``g = QgsGeometry( curveToAdd->clone() );`` at line 1390 ? I'm not familiar with ownerships of those object, but the lack of symmetry here is a suspicious, and if things are correct, this would call for a code comment explaining why things are handled differently

Reply:
It's closed in https://github.com/qgis/QGIS/blob/6029ea2219ee9fcae502cec380b5e3cb31691cb6/src/core/geometry/qgspolygon.cpp#L279-L281

Reply:
> It's closed in

perhaps I wasn't clear, but my question is about line 1419 ``g = QgsGeometry( poly );`` not doing a clone compared to line 1930 doing ``g = QgsGeometry( curveToAdd->clone() );``

Actually I see that the constructor is `` explicit QgsGeometry( QgsAbstractGeometry *geom SIP_TRANSFER );``, so I strongly suspect that line 1419  should be ``g = QgsGeometry( poly->clone() );``

Reply:
🤦  No, it's me... Sorry... I read "close" instead of "clone"...

Indeed, you're right. Fixed.